autopush/senderids.py

```diff
@@ -106,18 +108,15 @@ def update(self, senderIDs):
             # Skip using s3 (For debugging)
             if senderIDs:
                 self._senderIDs = senderIDs
-            self._updated = time.time()
             return
-        try:
-            bucket = self.conn.get_bucket(self.ID)
-            self._write(bucket, senderIDs)
-        except S3ResponseError:
-            self._create(senderIDs)
+        self._senderIDs = senderIDs
+        d = Deferred()
+        d.addCallback(self._write)
+        d.addErrback(self._create)
+        d.callback(senderIDs)
```

Comment:
This isn't any different than just running the functions in sequential order, since calling it with callback starts the chain running immediately.